### PR TITLE
Fix how to get kind from resource name

### PR DESCRIPTION
**Pull request by maruiz93**

### CODE REVIEW

**Comment by maruiz93** [comment]
Need to add tests

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -196,5 +196,6 @@ to start a debugger to which attach from your IDE.
    ```
    expose the `KIND_CLUSTER_NAME` env variable with the appropriate name of the kind cluster:
    ```bash
-   export KIND_CLUSTER_NAME=$(kubectl config get-clusters | grep -P '(?<=kind-).*' -o)
+   export KIND_CLUSTER_NAME=$(kind -q get clusters)
```
Comment (ggallen) [inline DEVELOPMENT.md:199]: How is this related to getting the kind from the resource name?

File: cmd/api/auth/authz.go [hunk 2]
```diff
@@ -7,6 +7,8 @@ import (
 	"fmt"
 	"net/http"
 
+	"github.com/kubearchive/kubearchive/cmd/api/abort"
```
Comment (ggallen) [inline cmd/api/auth/authz.go:10]: How are all these changes in authz.go related to getting the kind from the resource name?

File: cmd/api/auth/authn.go [hunk 3]
```diff
@@ -9,6 +9,8 @@ import (
 	"net/http"
 	"strings"
 
+	"github.com/kubearchive/kubearchive/cmd/api/abort"
```
Comment (ggallen) [inline cmd/api/auth/authn.go:12]: How are all these changes in authn.go related to getting the kind from the resource name?

File: cmd/api/routers/routers.go [hunk 4]
```diff
@@ -19,7 +23,7 @@ type Controller struct {
 func (c *Controller) GetAllResources(context *gin.Context) {
 	group := context.Param("group")
 	version := context.Param("version")
-	resource := getResourceKindFromPlural(context.Param("resourceType"))
+	resource := getAPIResourceKind(context)
```
Comment (ggallen) [inline cmd/api/routers/routers.go:26]: Shouldn't this really be called `kind`?

File: cmd/api/routers/routers.go [hunk 5]
```diff
@@ -33,7 +37,7 @@ func (c *Controller) GetNamespacedResources(context *gin.Context) {
 	group := context.Param("group")
 	version := context.Param("version")
 	namespace := context.Param("namespace")
-	resource := getResourceKindFromPlural(context.Param("resourceType")) + "/" + context.Param("namespace")
+	resource := getAPIResourceKind(context)
```
Comment (ggallen) [inline cmd/api/routers/routers.go:40]: Shouldn't this really be called `kind`?

**Review by ggallen — CHANGES_REQUESTED** [review]

File: DEVELOPMENT.md (hunk 1 above)
Comment (maruiz93) [reply DEVELOPMENT.md:199]: It is not, I just added the change because I faced problems when testing in my local with the other command.

It is ok for you to place it here or should I create separate commit in the PR? or a separate PR for this line?

I thought that with a separate PR it will be a little overkill as I guess I should create an issue for it.

File: cmd/api/auth/authn.go (hunk 3 above)
Comment (maruiz93) [reply cmd/api/auth/authn.go:12]: I created a new go module for the middleware layer in charge of doing the request of the api resources called ` discover`. This module needs the `Abort` function so I exported it outside from the `auth` module to its own one called `abort`.
Exporting the `abort` function and moving it to its own module created the need of importing it in the `auth` module to keep things working.

File: cmd/api/auth/authz.go (hunk 2 above)
Comment (maruiz93) [reply cmd/api/auth/authz.go:10]: See answer in the previous thread

**Review by skoved — APPROVED** [review]

File: cmd/api/routers/routers.go (hunk 5 above)
Comment (maruiz93) [reply cmd/api/routers/routers.go]: Fixed

File: cmd/api/routers/routers.go (hunk 4 above)
Comment (ggallen) [reply cmd/api/routers/routers.go]: No, I meant like this:
```
kind := getAPIResourceKind(context)
```

File: DEVELOPMENT.md (hunk 1 above)
Comment (ggallen) [reply DEVELOPMENT.md:199]: I'm OK with it, but I've seen on other projects that people complain when things are mixed.  I think this is mainly for reverting.  

If you do make additional changes, I tend to just note them in the description so that the reviewer is aware.

File: cmd/api/auth/authn.go (hunk 3 above)
Comment (ggallen) [reply cmd/api/auth/authn.go:12]: Understood.  I missed that link.

File: cmd/api/routers/routers.go (hunk 5 above)
Comment (ggallen) [reply cmd/api/routers/routers.go]: No, I meant like this:
```
kind := getAPIResourceKind(context)
```

File: cmd/api/routers/routers.go (hunk 4 above)
Comment (maruiz93) [reply cmd/api/routers/routers.go]: yes, that's the change that I did, don't you see it updated in the `Files Changed` section of this PR?
Note that this thread keeps the outdated code (there is the `outdated` label on it)

File: cmd/api/routers/routers.go (hunk 5 above)
Comment (maruiz93) [reply cmd/api/routers/routers.go]: see above comment, I think you aren't checking the most updated changes

**Review by ggallen — APPROVED** [review]
